Extension/src/packageVersion.ts

```diff
@@ -0,0 +1,56 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as telemetry from './telemetry';
+
+export class PackageVersion {
+    major: number;
+    minor: number;
+    patch: number;
+    suffix?: string;
+    isValid: boolean;
+
+    constructor(versionStr: string) {
+        let tokens: string[] = versionStr.split(new RegExp('[-\\.]', 'g')); // Match against dots and dashes
+        if (tokens.length < 3) {
+            this.isValid = false;
+            telemetry.logLanguageServerEvent('versionParsingFailure', { 'versionString': versionStr });
+            return;
+        }
+
+        this.major = parseInt(tokens[0]);
+        this.minor = parseInt(tokens[1]);
+        this.patch = parseInt(tokens[2]);
+        this.suffix = tokens[3];
```

Comment:
do you need to check the length of the array because grabbing the 4th element?

Reply:
`suffix` is marked optional, which according to [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html) means it is also an `undefined` type. There is no crash after assignment when `tokens[3]` evaluates to `undefined`, so my assumption is it's well formed. What do you think?